File: MapXtreme/6.6/Samples/WebControlsSourceCode/cs/MapControl/MapController.cs
using System;
using System.Drawing;
using System.Drawing.Imaging;
using System.IO;
using System.Runtime.Remoting.Contexts;
using System.Text;
using System.Web;
using System.Web.SessionState;
using System.Web.UI;
using System.Xml;

namespace MapInfo.WebControls
{
	/// <summary>
	/// Processes commands and generates a new image for the MapControl to display.
	/// </summary>
	/// <remarks>
	/// The controls from the client create a URL pointing to this handler with commands and data in it. 
	/// When the source of the image is set to this URL, this handler is called. The handler tries to get a model object from 
	/// the ASP.NET session. If it cannot get the model, it creates one and then calls the appropriate method based on the
	/// command it trying to process. After the model has performed the task, the image is exported and streamed back to client 
	/// so that only the map image gets updated. There is no need to do a postback of the entire page. 
	/// The model has the business logic to do the operation. Therefore, the controls
	/// are not dependant upon a particular product.
	/// </remarks>
	public class MapController : IHttpHandler, IRequiresSessionState {
		/// <summary>
		/// Reuses the instance of this handler.
		/// </summary>
		/// <value>Returns <c>true</c>.</value>
		/// <remarks>None</remarks>
		public bool IsReusable {
			get {
				return true;
			}
		}

		/// <summary>
		/// Executes commands and delegates the execution of the command to the model.
		/// </summary>
		/// <remarks>The model extracts the command and calls the appropriate method to perform the operation.
		/// See description of the class for more details.</remarks>
		/// <param name="context">Current context</param>
		public void ProcessRequest(HttpContext context) {
			MapControlModel mapModel = MapControlModel.SetDefaultModelInSession();
			if (mapModel != null) {
				mapModel.InvokeCommand();
			}
		}
	}
}

File: trunk/项目开发进度2009年2月9日/SeeMapCell/MapXtreme/6.6/Samples/WebControlsSourceCode/vb/MapControl/MapController.vb
Imports System
Imports System.Collections
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.IO
Imports System.Web
Imports System.Web.SessionState
Imports System.Web.UI
Imports System.Web.UI.WebControls

'/ <summary>
'/ This handler's job is to process command and generate new image for MapControl to display
'/ </summary>
'/ <remarks>
'/ The controls from the client create a url pointing to this handler with command and data in it. When the source of the image is set to this url,
'/ this handler is called. Handler tries to get model object from ASP.NET session, if not creates one and then calls appropriate method based on the
'/ command it trying to process. After the model has done it's job the image is exported and streamed back to client and thus only the map image 
'/ gets updated and there is no need to do postback of the entire page. The model has the actual business logic to do operation, therefore the controls
'/ are not dependant upon any product.
'/ </remarks>
Public Class MapController
    Implements IHttpHandler, IRequiresSessionState
    '/ <summary>
    '/ Constructor for the handler
    '/ </summary>
    '/ <remarks>None</remarks>
    Public Sub New()
    End Sub

    '/ <summary>
    '/ Resue the instance of this handler
    '/ </summary>
    '/ <value>Returns true</value>
    '/ <remarks>None</remarks>
    Public ReadOnly Property IsReusable() As Boolean Implements IHttpHandler.IsReusable
        Get
            Return True
        End Get
    End Property

    '/ <summary>
    '/ This method executes commands. It delegates the execution of the command to Model. The Model extracts the command and calls appropriate 
    '/ method to do the operation
    '/ </summary>
    '/ <remarks>See description of the class for more details</remarks>
    '/ <param name="context">Current context</param>
    Public Sub ProcessRequest(ByVal context As HttpContext) Implements IHttpHandler.ProcessRequest
        Dim mapModel As MapControlModel = MapControlModel.SetDefaultModelInSession()
        If Not mapModel Is Nothing Then
            mapModel.InvokeCommand()
        End If
    End Sub
End Class

